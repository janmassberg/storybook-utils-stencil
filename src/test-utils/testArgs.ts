export const testArgs = {
    testNull: null,
    testUndefined: undefined,
    testFalse: false,
    testTrue: true,
    testString: "test-string",
    testNumber: 12345,
    testArray: [1, 2, 3],
    testObject: {
        foo: "bar",
    },
    testFunction: (): void => {
        // Implementation
    },
    onTestEvent: (event: Event): void => {
        // Handle event
        console.log(event.type);
    },
    _slot: "Test children",
};

export const testArgsButton = {
    required: true,
    variant: "primary",
    onClick: (event: Event) => {
        // Handle click event
        console.log(event.target);
    },
    _slot: "Label",
};

export const testArgsComponentGenerator = {
    component: "my-component",
    args: testArgs,
};

export const testCopmonentSlot = `<my-child-component foo-bar="foo">
<my-nested-child-component bar-foo="bar"><span class="test-class">Label</span></my-nested-child-component>
</my-child-component>`;
