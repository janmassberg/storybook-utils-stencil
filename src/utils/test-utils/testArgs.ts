export const testArgs = {
    testNull: null,
    testUndefined: undefined,
    testFalse: false,
    testTrue: true,
    testString: "test-string",
    testNumber: 12345,
    testArray: [1, 2, 3],
    testObject: {
        foo: "bar",
    },
    testFunction: () => {},
    onTestEvent: () => {},
    _children: "Test children",
};
