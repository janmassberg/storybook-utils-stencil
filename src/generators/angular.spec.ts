import {
    generateSourceCodeAngularComponent,
    generateSourceCodeAngularHtml,
} from "./angular";
import { testArgsComponentGenerator, testCopmonentSlot } from "../test-utils/testArgs";

describe("generateSourceCodeAngularComponent", () => {
    it("should generate valid formatted angular component source code", () => {
        expect(
            generateSourceCodeAngularComponent(testArgsComponentGenerator)
        ).toMatchSnapshot();
    });

    it("should generate valid formatted angular component source code with children", () => {
        expect(
            generateSourceCodeAngularComponent({
                ...testArgsComponentGenerator,
                args: {
                    _slot: testCopmonentSlot,
                }
            })
        ).toMatchSnapshot();
    });
});

describe("generateSourceCodeAngularHtml", () => {
    it("should generate valid formatted angular html template source code", () => {
        expect(
            generateSourceCodeAngularHtml(testArgsComponentGenerator)
        ).toMatchSnapshot();
    });

    it("should generate valid formatted angular html template source code with children", () => {
        expect(
            generateSourceCodeAngularHtml({
                ...testArgsComponentGenerator,
                args: {
                    _slot: testCopmonentSlot,
                }
            })
        ).toMatchSnapshot();
    });
});
