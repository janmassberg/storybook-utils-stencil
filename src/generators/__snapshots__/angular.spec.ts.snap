// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSourceCodeAngularComponent should generate valid formatted angular component source code 1`] = `
"import { Component } from "@angular/core";

@Component({
  selector: "app-my-component-example",
  templateUrl: "./my-component-example.component.html",
})
export class MyComponentExampleComponent {
  public testArray = [1, 2, 3];

  public testObject = { foo: "bar" };

  public testFunction = () => {
    // Implementation
  };

  onTestEvent(event: Event) {
    // Handle event
    console.log(event.type);
  }
}
"
`;

exports[`generateSourceCodeAngularComponent should generate valid formatted angular component source code with children 1`] = `
"import { Component } from "@angular/core";

@Component({
  selector: "app-my-component-example",
  templateUrl: "./my-component-example.component.html",
})
export class MyComponentExampleComponent {}
"
`;

exports[`generateSourceCodeAngularHtml should generate valid formatted angular html template source code 1`] = `
"<my-component
  test-false="false"
  test-true="true"
  test-string="test-string"
  test-number="12345"
  [testArray]="testArray"
  [testObject]="testObject"
  [testFunction]="testFunction"
  (testEvent)="onTestEvent($event)"
>
  Test children
</my-component>
"
`;

exports[`generateSourceCodeAngularHtml should generate valid formatted angular html template source code with children 1`] = `
"<my-component>
  <my-child-component foo-bar="foo">
    <my-nested-child-component bar-foo="bar">
      <span class="test-class">Test children</span>
    </my-nested-child-component>
  </my-child-component>
</my-component>
"
`;
